packages/nimble-components/src/table/index.ts

```diff
@@ -409,6 +543,21 @@ export class Table<
             };
         });
     }
+
+    private calculateTanStackSelectionState(
+        recordIds: string[]
+    ): TanStackRowSelectionState {
+        const tanstackSelectionState: TanStackRowSelectionState = {};
+        for (const id of this.tableValidator.getPresentRecordIds(recordIds)) {
```

Comment:
I wrote it in preparation for multiple selection, but since we don't have that yet, I'll update the implementation.

I think the name `selectedRecordIds` is a bit misleading because they aren't actually selected at the point the function is called. I'll rename it to `recordIdsToSelect` instead.